pkg/pillar/types/certinfotypes.go

```diff
@@ -25,3 +55,22 @@ type ControllerCert struct {
 func (cert *ControllerCert) Key() string {
 	return hex.EncodeToString(cert.CertHash)
 }
+
+// DeviceCert : device certificate
+// device generated certificate
+type DeviceCert struct {
```

Comment:
This can use the existing AttestCert definition. It already has a CertType. Just need to add a bool for InTPM to that type.